mp/src/game/shared/neo/weapons/weapon_knife.cpp

```diff
@@ -290,9 +290,24 @@ void CWeaponKnife::Hit(trace_t& traceHit, [[maybe_unused]] Activity nHitActivity
 		VectorNormalize(hitDirection);
 
 #ifndef CLIENT_DLL
-		CTakeDamageInfo info(GetOwner(), GetOwner(), KNIFE_DAMAGE, DMG_SLASH);
 
-		CalculateMeleeDamageForce(&info, hitDirection, traceHit.endpos);
+		Vector forward;
+		AngleVectors(pHitEntity->GetAbsAngles(), &forward);
+
+		Vector2D forward2D = Vector2D(forward.x, forward.y);
+		forward2D.NormalizeInPlace();
+		
+		Vector attackerToTarget = pHitEntity->GetAbsOrigin() - pPlayer->GetAbsOrigin();
+		Vector2D attackerToTarget2D = Vector2D(attackerToTarget.x, attackerToTarget.y);
+		attackerToTarget2D.NormalizeInPlace();
+
+		float dotProduct = acos(DotProduct2D(forward2D, attackerToTarget2D));
+
+		//engine->Con_NPrintf(3, "Angle: %f", dotProduct);
+
+		CTakeDamageInfo info(GetOwner(), GetOwner(), (dotProduct > 0.6435 ? KNIFE_DAMAGE : (100 * (1 / NEO_SUPPORT_DAMAGE_MODIFIER)) + 1), DMG_SLASH);
+
```

Comment:
https://github.com/NeotokyoRebuild/neo/assets/44210793/062f6661-a54b-43d9-b1b0-a2d2ff261519

I'm struggling to replicate this. I have un-commented out the angle readout in the top right corner, notice how when I set the yaw offset to 36.86 the angle in the top right is < 0.6345, and for an angle of 36.87 the angle is > 0.6345. I'm rotating the bots to a large angle between small changes to force a rotation, you can also wait a few seconds and their lower body should turn to face the angle in which they are looking. I also freeze bot_think so that they do not mimic my attack which over-writes the readout since the bots are also attacking with a melee weapon, but from testing off camera (10MB recording limit) this doesn't seem to impact the results

Reply:
I just re-read your post and you mention an angle of 36.87 results in a backstab but an angle of 36.88 doesn't. If this is what the issue is about, that's because I didn't catch the second message on the angle that you sent, so was still using this value 
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/1db8c18c-e0ef-49d1-b726-03e89918b5f7)

Reply:
@AdamTadeusz I could've made a mistake with my testing yesterday, I'll try again.

Reply:
@AdamTadeusz Ok, tested again and the angle is correct. I must've messed something up with my angles yesterday, sorry. But could update the angle to `0.6435011` (constexpr estimation of `math.asin(0.6)`), so it's as accurate as it can be?